### PR TITLE
Run common tasks only on install/update

### PR DESCRIPTION
**Pull request by scollazo**
They install/update packages, and that slows ansible runs
used for configuration only (_archivematica_src_install_am: 'false'_ and _archivematica_src_install_ss: 'false'_ )

### CODE REVIEW

**Comment by luisilara** [comment]
LGTM

**Review by DavidHume-ArtefactualSystems — COMMENTED** [review]
Possibly nitpick: should not the **-name** sections be indented a couple of spaces in the **block:**? I could swear that was important and would break (not just about readability) in ansible playbooks, maybe that's not universal.

Interesting, I gather these values potentially being "rpm" besides "yes", "True" or etc. replaces briefly considered other boolean variables like _archivematica_src_install_am_rpm:_? Not sure if I would have made that choice stylistically but should work, and kinda slick.

If this role has already been actually tested in all conditions (e.g. 'rpm' and 'true' and false/undefined) thus rendering these questions moot, consider this approval.

File: tasks/main.yml [hunk 1]
```diff
@@ -53,26 +53,37 @@
...
+  - name: "Include common-RHEL.yml for RHEL"
+    include: "common-RHEL.yml"
+    when:
+      - "ansible_distribution == 'RedHat'"
+
+  - name: "Install necessary packages required by this role"
+    package:
+      name: "{{ item }}"
+      state: "latest"
+    with_items: "{{ ansible_deps }}"
```
Comment (mamedin) [inline tasks/main.yml:69]: We should stop using `with_items` (deprecated) in the apt, yum and packages modules whenever possible. In the case of osdeps where the state is included in the dict, it is more complicated to do, but for these simple cases we should use:

```
  - name: "Install necessary packages required by this role"
    package:
      name: "{{ ansible_deps }}"
      state: "latest"
``` 

And when using a explicit list of packages:

``` 
   - name: "Install necessary packages required by rpms"
     yum:
       name:
         - "python-virtualenv"
         - "python-pip"
       state: "present"
```

Instead of:

```
   - name: "Install necessary packages required by rpms"
     yum:
       name: "{{ item }}"
       state: "present"
     with_items:
       - "python-virtualenv"
       - "python-pip"
``` 

We can change it here or make a separate PR.

**Review by mamedin — APPROVED** [review]
LGTM

**Comment by scollazo** [comment]
@mamedin that wasn't on the scope of this PR, so I'm merging it as is and leaving the fix for another pr
